instrumentation/spring/spring-webflux/spring-webflux-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webflux/v5_0/server/HandlerAdapterInstrumentation.java

```diff
@@ -75,40 +105,30 @@ public static void methodEnter(
           parentContext, HttpServerRouteSource.NESTED_CONTROLLER, httpRouteGetter(), exchange);
 
       if (handler == null) {
-        return;
+        return null;
       }
 
       if (!instrumenter().shouldStart(parentContext, handler)) {
-        return;
+        return null;
       }
-
-      context = instrumenter().start(parentContext, handler);
-      scope = context.makeCurrent();
+      Context context = instrumenter().start(parentContext, handler);
+      return new AdviceScope(context, context.makeCurrent());
```

Comment:
Any reason for not using the pattern of a `static enter()` method on `AdviceScope` here instead of having the logic in the Advice?